### PR TITLE
Actions in narrative view

### PR DESCRIPTION
**Pull request by jessica-oflanagan**
_Pull requests into Flux require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:._

Moved the logic for filtering/displaying actions for a subsection into one file, VisualizerMenu.  VisualizerMenu now takes a subsection and a list of actions, and filters and displays them according to their whenToDisplay properties.

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome
- Manually tested in IE

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- Cheat sheet is updated
- Demo script is updated 
- Documentation on Wiki has been updated 
- Additional technical components and libraries are documented and added to wiki as needed

**NoteParser**

- Note parser has been updated if structured phrases change

**Code Quality**

- [x] 4-space indents - convert any tabs to spaces
- [x] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- Added Enzyme-UI tests for slim mode 
- Added Enzyme-UI tests for full mode
- Added backend tests for new functionality added
- Added note parser examples to test new functionality

## Reviewer 1: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: src/summary/NarrativeNameValuePairsVisualizer.jsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
-import { ListItemIcon, ListItemText } from 'material-ui/List';
-import Menu, { MenuItem } from 'material-ui/Menu';
-import FontAwesome from 'react-fontawesome';
+//import { ListItemIcon, ListItemText } from 'material-ui/List';
```
[the remark this answers is not in the document]
Comment (jessica-oflanagan) [reply src/summary/NarrativeNameValuePairsVisualizer.jsx]: All set!

**Comment by jafeltra** [comment]
This will just need to be rebased!
